astronomer/providers/google/cloud/extractors/bigquery_async_extractor.py

```diff
@@ -0,0 +1,141 @@
+import json
+import logging
+from typing import Any, List, Optional, Union
+
+import attr
+from airflow.exceptions import AirflowException
+from airflow.models.taskinstance import TaskInstance
+from airflow.utils.log.logging_mixin import LoggingMixin
+from google.cloud.bigquery import Client
+from openlineage.airflow.extractors.base import BaseExtractor, TaskMetadata
+from openlineage.airflow.utils import get_job_name
+from openlineage.client.facet import SqlJobFacet
+from openlineage.common.provider.bigquery import BigQueryDatasetsProvider
+from openlineage.common.sql import parse
+
+from astronomer.providers.google.cloud.hooks.bigquery import _BigQueryHook
+from astronomer.providers.google.cloud.operators.bigquery import (
+    BigQueryCheckOperatorAsync,
+    BigQueryGetDataOperatorAsync,
+    BigQueryInsertJobOperatorAsync,
+    BigQueryIntervalCheckOperatorAsync,
+    BigQueryValueCheckOperatorAsync,
+)
+
+log = logging.getLogger(__name__)
+
+
+@attr.s
+class SqlContext:
+    """Internal SQL context for holding query parser results"""
+
+    sql: str = attr.ib()
+    inputs: Optional[str] = attr.ib(default=None)
+    outputs: Optional[str] = attr.ib(default=None)
+    parser_error: Optional[str] = attr.ib(default=None)
+
+
+class BigQueryAsyncExtractor(BaseExtractor, LoggingMixin):
```

Comment:
BaseExtractor is should have derived from LoggingMixin, if that case, then BigQueryAsyncExtractor need not to derived from LoggingMixin. Can you please check?

Reply:
right, removed.